game-core/src/main/java/games/strategy/engine/lobby/server/User.java

```diff
@@ -3,60 +3,27 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 
 import java.net.InetAddress;
-import java.util.Objects;
 
 import javax.annotation.concurrent.Immutable;
 
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
 /**
  * Information about a lobby user.
  */
 @Immutable
+@Builder
+@Value
 public final class User {
+  @NonNull
   private final String hashedMacAddress;
+  @NonNull
   private final InetAddress inetAddress;
+  @NonNull
   private final String username;
 
-  public User(final String username, final InetAddress inetAddress, final String hashedMacAddress) {
```

Comment:
I think the tests are failing because the generated all-args constructor is probably using field order for the parameters, which would be `hashedMacAddress`, `inetAddress`, and `username`.  That means `hashedMacAddress` and `username` are being reversed wherever the `User` constructor is being called directly.

So not sure if we should disable the all-args constructor generation and just require builder usage everywhere?  Or re-order the fields?  (Reordering the fields seems a bit fragile, but maybe that's just the Lombok Way, and we have to get used to it.)

Reply:
> but maybe that's just the Lombok Way, and we have to get used to it.

Lombok is not opinionated on how you use it. Side by side String params is a problem if you use all-args or not, you still have the problem, it has just been highlighted now that it's easy to swap the ordering of parameters so easily.

@ssoloff good investigation on test failure, thank you for point me to the root cause. Parameter swapping is still just as fragile before, I think the option you pointed to do forcing builder everywhere is the way to go. I did not realize the param ordering had been swapped, that was a mistake. Regardless, forcing builder will help reduce the vulnerability where parameters can be swapped so easily.

Reply:
> but maybe that's just the Lombok Way, and we have to get used to it.

I've a strong reaction to this. Lombok magnifies existing code practices, I don't think there is anything intrinsic to it necessarily, it is not opinionated, just how you use it (in this case without enough thought; but also I tend to rely on tests a bit too much to move faster than i should)